runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -288,7 +299,7 @@ public int appendObjectState(Stream<Map.Entry> entryStream) {
              * of SMR entries. Its a safeguard against the smr entries amounting to the actual
              * boundary limit.
              */
-            if (numBytesPerCheckpointEntry > maxWriteSizeLimit) {
+            if (numBytesPerCheckpointEntry > maxWriteSizeLimit || smrEntries.getUpdates().size() >= batchSize) {
```

Comment:
Why don't you just let it fill the maxWriteSizeLimit buffer, but remove the compression from the calculation? I think that's cleaner. 

The batch size parameter needs to be tuned to match the max write limit, if you just base your calculation on the uncompressed size then you can get rid of this tuning problem all together.

Reply:
I agree that comparing serialized but uncompressed size could help stay away from such OOM headache. The original idea to calculate the compressed size is because we want to batch as much as possible which in theory could provide some performance improvement.

Reply:
The initial problem - as I remember it - was that using number of entries to determine batch size was generating LogData entries that were to large to send over RPC. The initial solution shouldn't have used the compressed bytes in its calculation.

If the uncompressed size can fit in an RPC message, and a client is able to receive this message, then it can be sure that what ever is decompressed will be smaller than the largest message it can receive.